### PR TITLE
Missing setup to work with Azure OpenAI

### PR DESCRIPTION
**Pull request by flanza**
Thank you for building this amazing library!
This PR has my 50-cent contribution. It contains the changes I had to do to make it actually work with Azure OpenAI.
It would be great to see it merged so I can point my project back to the main Maven repo.

### CODE REVIEW

File: langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java [hunk 1]
```diff
@@ -68,8 +71,15 @@ public OpenAiChatModel(String baseUrl,
         timeout = timeout == null ? defaultTimeoutFor(modelName) : timeout;
         maxRetries = maxRetries == null ? 3 : maxRetries;
 
-        this.client = OpenAiClient.builder()
-                .openAiApiKey(apiKey)
+        final OpenAiClient.Builder builder = OpenAiClient.builder();
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java:74]: please check `AzureOpenAi*Model` classes

**Review by ghost — COMMENTED** [review]
Hi @flanza thank you for your contribution! But for Azure, we have a separate set of `AzureOpenAi*Model` classes, please check those

File: langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java (hunk 1 above)
Comment (flanza) [reply langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java:74]: Hi @langchain4j,

Thank you for pointing that out. Initially, I was trying to use this set of classes, but I came across [this comment](https://github.com/langchain4j/langchain4j/blob/main/langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java#L23) saying function calling is not supported in `AzureOpenAiLanguageModel`, and this is a key part of what I'm trying to accomplish.

This is when I decided to add the missing code to set up my Azure OpenAI along with `OpenAi*Model` classes and create the PR. 

Best,
Fabio

Comment (langchain4j) [reply langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java:74]: @flanza `AzureOpenAiLanguageModel` does not support function calling, but `AzureOpenAiChatModel` does, as written in the comment.

Comment (flanza) [reply langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java:74]: Cool! In this case, I will try it and let you know the outcome. If it works smoothly we can close this PR without merging.

Thank you again for pointing this out 👍

**Comment by langchain4j** [comment]
@flanza closing this due to inactivity. please feel free to reopen if needed

**Pull request closed without merging**